docker/puppetdb/docker-entrypoint.d/00-envsubst.sh

```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
+
+# NOTE: define ENV variables and defaults in the Dockerfile
+
+envsub() {
+    envsubst < "$1" > "${1}.new"
+    mv --force "${1}.new" "$1"
+}
+
+for f in /etc/puppetlabs/puppetdb/conf.d/*.conf; do
```

Comment:
I don't think we even need to do this, do we? 

I think `database.conf` already understands / uses the env vars defined in the environment, doesn't it?

Reply:
You'd think it would work but it didn't when I tried it before submitting the PR.

The existence of quotes screws it up, and without them the syntax is incorrect.

Reply:
```
subname: \"//${PUPPETDB_POSTGRES_HOSTNAME}:${PUPPETDB_POSTGRES_PORT}/${PUPPETDB_POSTGRES_DATABASE}\"
```

Error:
```

Reply:
That reads to me like the `\`'s aren't necessary, does that work?

Reply:
I'm reproducing all the ways it fails....stay tuned

Reply:
Error:

Reply:
```
  subname: //${PUPPETDB_POSTGRES_HOSTNAME}:${PUPPETDB_POSTGRES_PORT}/${PUPPETDB_POSTGRES_DATABASE}
```

Error:
```
puppetdb_1  | Exception in thread "main" com.typesafe.config.ConfigException$Parse: /etc/puppetlabs/puppetdb/conf.d/database.conf: 3: Expecting close brace } or a comma, got ':' (if you intended ':' to be part of the value for 'subname', try enclosing the value in double quotes)
```

Reply:
Ok...see above for the 3 forms that don't work.

Reply:
The behavior of this .conf file when it's read by PuppetDB seems to be:
- Quotes prevent variable interpolation
- Unquoted is a syntax error (for this particular value)

Reply:
:sigh: computers are the worst

Reply:
Should we file a ticket / leave a comment here about why the typical expected PDB behavior doesn't work?

/cc @rbrw @robdaemon @austb

Reply:
file a ticket, please :)

Reply:
It looks like this is a HOCON thing per https://github.com/lightbend/config/issues/633

Reply:
So intuitive... 🤦‍♂

Reply:
Nice find. That works!